.github/integration-in-memory-tests.yml

```diff
@@ -15,4 +15,12 @@ runner-test-matrix:
     triggers:
       - PR Integration CCIP Tests
     test_cmd: cd integration-tests/smoke/ccip && go test ccip_messaging_test.go -timeout 12m -test.parallel=2 -count=1 -json
+
+  - id: contracts/ccipreader_test.go:*
+    path: integration-tests/contracts/ccipreader_test.go
+    test_env_type: in-memory
+    runs_on: ubuntu-latest
+    triggers:
+      - PR Integration CCIP Tests
+    test_cmd: cd integration-tests/contracts && go test ccipreader_test.go -timeout 12m -test.parallel=1 -count=1 -json
```

Comment:
Right now, this test matrix runs within 8-9 mins but we can play with parallel count and see if we can speed up. Not in this PR but in follow up PR.

Reply:
Wait, the whole matrix takes 8-9 minutes or just the ccip reader tests? That seems awfully long, it should be < 1 min for the whole suite

Reply:
One of the test (TestCCIPReader_CommitReportsGTETimestamp_RespectsFinality) runs for more than 5mins. 
<img width="677" alt="Screenshot 2024-11-28 at 12 14 51 PM" src="https://github.com/user-attachments/assets/c9222bc5-563b-4105-aced-15d9a49d9382">

DB setup takes 3 mins.